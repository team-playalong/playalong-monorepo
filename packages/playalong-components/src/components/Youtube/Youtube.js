import * as React from 'react';
import 'lazysizes/lazysizes.min.js';
import PropTypes from 'prop-types';

Youtube.propTypes = {
  id: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
  videoId: PropTypes.string.isRequired,
}

const defaultProps = {
  id: 'plyYoutubeIframe',
  width: 300,
  height: 150,
};

const youtubePrefix = 'https://www.youtube.com/embed/';

function buildVideoId(videoId) {
  return youtubePrefix + videoId;
}

function Youtube({
  id = defaultProps.id,
  width = defaultProps.width,
  height = defaultProps.height,
  videoId,
}) {

  return (
    <iframe
      title="playalong youtube"
      id={id}
      className={'lazyload'}
      width={width}
      height={height}
      data-src={buildVideoId(videoId)}
      frameBorder='0'
      allowFullScreen
    >
    </iframe>
  );
}

export const props = ['id', 'width', 'height', 'videoId'];
export default Youtube;
