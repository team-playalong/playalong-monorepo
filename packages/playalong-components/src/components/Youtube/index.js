export {default, props} from './Youtube';
