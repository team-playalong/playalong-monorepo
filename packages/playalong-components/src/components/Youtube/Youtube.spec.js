import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Youtube from './Youtube';

let wrapper;
const props = {
  videoId: '123',
  width: undefined,
  height: undefined,
};

beforeEach(() => {
  wrapper = shallow(<Youtube {...props} />);
});

test.skip('Should have a src', () => {
  expect(wrapper.find('iframe').props().src).toContain(props.videoId);
});

test('Should have default width and height', () => {
  expect(wrapper.find('iframe').props().width).toBe(300);
  expect(wrapper.find('iframe').props().height).toBe(150);
});


test.skip('Snapshot testing', () => {
  const tree = renderer.create(<Youtube
    videoId={'123'} />).toJSON();

  expect(tree).toMatchSnapshot();
});
