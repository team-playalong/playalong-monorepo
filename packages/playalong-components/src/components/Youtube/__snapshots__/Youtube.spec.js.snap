// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<iframe
  allowFullScreen={true}
  className="lazyload"
  frameBorder="0"
  height={150}
  id="plyYoutubeIframe"
  src="https://www.youtube.com/embed/123"
  title="playalong youtube"
  width={300}
/>
`;
